journal_brief/journal_brief.py

```diff
@@ -153,10 +153,24 @@ def __init__(self, cursor_file=None, reader=None, dry_run=False,
         if reader is None:
             reader = journal.Reader()
 
-        if seek_cursor and self.cursor:
-            log.debug("Seeking to %s", self.cursor)
-            reader.seek_cursor(self.cursor)
-            reader.get_next()
+        if self.cursor:
+            if seek_cursor:
+                log.debug("Seeking to %s", self.cursor)
+                reader.seek_cursor(self.cursor)
+                reader.get_next()
+        else if not dry_run:
+            # use an unfiltered Reader to find the current 'tail'
+            # of the journal and store that as the initial cursor
+            # when the cursor file could not be found; this avoids
+            # reading through the entire journal again on the next
+            # run if the inclusions and exclusions result in zero
+            # matching entries during this run
+            temp_reader = journal.Reader()
+            temp_reader.seek_tail()
+            fields = temp_reader.get_previous()
+            if fields:
+                self.cursor = fields['__CURSOR']
```

Comment:
Probably still need to fall back to `''` if there are no journal entries at all. That's a real corner case! Could do this by initialising to `''` in the `except` clause above.